ovs/lib/helpers/vpool/shared.py

```diff
@@ -38,3 +43,79 @@ def retrieve_alba_arakoon_config(alba_backend_guid, ovs_client):
+        """
+        cfg = Configuration.get(REMOTE_CONFIG_BACKEND_CONFIG.format(alba_backend_guid), default=None)
+        if as_ini:
+            cfg = ArakoonClusterConfig.convert_config_to(cfg, return_type='INI')
+        return cfg
+
+    @classmethod
+    def sync_alba_arakoon_config(cls, alba_backend_guid, ovs_client):
+        # type: (str, OVSClient) -> None
+        """
+        Compares the remote and local config. Updates the local config if needed. Guarantees the latest greatest config
+        WARNING: YOU DO NOT BELONG HERE, PLEASE MOVE TO YOUR OWN PLUGIN
+        :param alba_backend_guid: Guid of the ALBA Backend
+        :type alba_backend_guid: str
+        :param ovs_client: OVS client object
+        :type ovs_client: OVSClient
+        :return: Arakoon configuration information
+        :rtype: dict
+        """
+        #todo: Fix leak of backends not bein removed
+
+        remote_config = cls._retrieve_remote_alba_arakoon_config(alba_backend_guid, ovs_client)
+        current_config = Configuration.get(REMOTE_CONFIG_BACKEND_CONFIG.format(alba_backend_guid), default=None)
+        if current_config != remote_config:
+            transaction_id = Configuration.begin_transaction()
+            Configuration.set(key=REMOTE_CONFIG_BACKEND_CONFIG.format(alba_backend_guid), value=remote_config, transaction=transaction_id)
+
+            ini_config = ArakoonClusterConfig.convert_config_to(config=remote_config, return_type='INI')
+            Configuration.set(key=REMOTE_CONFIG_BACKEND_INI.format(alba_backend_guid), value=ini_config, raw=True, transaction=transaction_id)
+            Configuration.apply_transaction(transaction_id)
+
+
+    @classmethod
+    def calculate_abm_configs_in_use(cls):
+        """
+        Iterate over all vPools in the cluster and check which local or remote backend configs are still in use.
+        :return:
+        """
+        present_remote_configs = dict([(key, REMOTE_CONFIG_BACKEND_CONFIG.format(key)) for key in list(Configuration.list(REMOTE_CONFIG_BACKEND_BASE))])
+        in_use = set()
+        cache_types = [StorageDriverConfiguration.CACHE_FRAGMENT, StorageDriverConfiguration.CACHE_BLOCK]
+
+        for vpool in VPoolList.get_vpools():
+            print vpool.name
+            proxy_config_template = '/ovs/vpools/{0}/proxies/{{0}}/config/main'.format(vpool.guid)
+            for std in vpool.storagedrivers:
+                for proxy in std.alba_proxies:
+                    cfg = Configuration.get(proxy_config_template.format(proxy.guid), default=None)
+                    cfg_mgr_url = cfg['albamgr_cfg_url']  # type:str
+
+                    if cfg_mgr_url not in present_remote_configs.values():
+                        print cfg_mgr_url
+                        in_use.add(Configuration.extract_key_from_path(cfg_mgr_url))
+                    # Extract all albamgr_cfg_urls
+                    for cache_type in cache_types:
+                        cache_cfg = cfg.get(cache_type)
```

Comment:
If you want to ensure that you get a list, use .get(key, [])
len(None) -> error